packages/app/lib/src/features/home/data/post_repository.dart

```diff
@@ -0,0 +1,101 @@
+import 'package:appwrite/appwrite.dart';
+import 'package:appwrite/models.dart';
+import 'package:flutter/material.dart';
+import 'package:latlong2/latlong.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+
+import '../../../../env/env.dart';
+import '../../../utils/api.dart';
+import '../domain/post_entity.dart';
+
+part 'post_repository.g.dart';
+
+/*
+How to do things:
+Put database and collection id in .env
+Implement post code from other branch
+For first implementation have post_repository read in all posts and then add queries and other filters down the line
+Have flutter access user location
+*/
+
+// Might need to have 2 posts lists. One for local and one for world
+
+// Post_entitys are put into list in feed_service
+
+abstract interface class PostRepository {
+  //Have funcs in heres
+}
+
+final class _AppwritePostRepository implements PostRepository {
+  String databaseId = Env.databaseId;
+  String collectionId = Env.postsCollectionId;
+
+  final Databases database = Databases(client as Client);
+
+  List<Document> _posts = [];
+
+  //get all todos func for other classes
+  List<Document> get allPosts => _posts;
+
+  bool _isLoading = false;
+
+  //check if loading
+
+  bool get checkLoading => _isLoading;
+
+  // Read all the todos
+
+  Future readPosts() async {
+    var isLoading = true;
+    try {
+      final data = await database.listDocuments(
+        databaseId: databaseId,
+        collectionId: collectionId,
+        // queries: [
+        //   Query.equal('createdBy', email),
+        // ],
+      ); // Only returns data with these attributes
+      _posts = data.documents;
+      isLoading = false;
+      // notifyListeners();
+    } catch (e) {
+      _isLoading = false;
+      // notifyListeners();
+      print(e);
+    }
+  }
+
+  //Create a new Todo
+
+  Future createNewPost(
+    String? headline,
+    String? description,
+    UserId author,
+    LatLng location,
+    ImageProvider? image,
+  ) async {
+    // Maybe change to no author param and getting value in here. Whatever's more efficient
+    //TODO get current users email or user ID (talk to eli about which one)
```

Comment:
If the author is always current user, then yes. We probably want to cache it though, so no?
I recall changing my mind in [`PSDTools/app`](https://github.com/PSDTools/app), so check the [`CoinsRepository`](https://github.com/PSDTools/app/blob/main/lib/features/pirate_coins/data/coins_repository.dart) over there.

Reply:
Answer: I ended up passing it in as an argument to the repo. Note that to do that, you'd want to get rid of state in the repo (which is necessary either way) so that you don't run the risk of a singleton.